packages/btcindexer/src/btcindexer.ts

```diff
@@ -326,4 +368,85 @@ export class Indexer {
 		}
 		return updates;
 	}
+	async callSuiMintContract(
+		transaction: Transaction,
+		blockHeight: number,
+		txIndex: number,
+		proof: ProofResult,
+	): Promise<boolean> {
+		try {
+			const tx = new SuiTransactoin();
+			const target = `${this.suiPackageId}::nbtc::mint_nbtc`;
+			tx.moveCall({
+				target: target,
+				arguments: [
+					tx.object(this.suiNbtcObjectId),
+					tx.object(this.suiLightClientObjectId),
+					tx.pure.vector("u8", this.serializeU32(transaction.version)),
+					tx.pure.u32(transaction.ins.length),
+					tx.pure.vector("u8", this.serializeTxInputs(transaction.ins)),
+					tx.pure.u32(transaction.outs.length),
+					tx.pure.vector("u8", this.serializeTxOutputs(transaction.outs)),
+					tx.pure.vector("u8", this.serializeU32(transaction.locktime)),
+					tx.pure.vector(
+						"vector<u8>",
+						proof.proofPath.map((p) => Array.from(p)),
+					),
+					tx.pure.u64(blockHeight),
+					tx.pure.u64(txIndex),
+				],
+			});
+
+			const result = await this.suiClient.signAndExecuteTransaction({
+				signer: this.suiSigner,
+				transaction: tx,
+				options: {
+					showEffects: true,
+				},
+			});
+
+			if (result.effects?.status.status === "success") {
+				console.log(`Mint successful. Digest: ${result.digest}`);
+				return true;
+			} else {
+				console.error(
+					`Mint failed. Digest: ${result.digest}:`,
+					result.effects?.status.error,
+				);
+				return false;
+			}
+		} catch (error) {
+			console.error(`Error during mint call`, error);
+			return false;
+		}
+	}
+
+	serializeU32(n: number): number[] {
+		const buffer = Buffer.alloc(4);
+		buffer.writeUInt32LE(n, 0);
+		return Array.from(buffer);
+	}
+
+	serializeTxInputs(inputs: Input[]): number[] {
+		const buffers = inputs.map((vin) => {
+			const hash = Buffer.from(vin.hash);
+			const index = Buffer.alloc(4);
+			index.writeUInt32LE(vin.index, 0);
+			const scriptLen = Buffer.from([vin.script.length]);
+			const sequence = Buffer.alloc(4);
+			sequence.writeUInt32LE(vin.sequence, 0);
+			return Buffer.concat([hash, index, scriptLen, vin.script, sequence]);
+		});
+		return Array.from(Buffer.concat(buffers));
+	}
+
+	serializeTxOutputs(outputs: Output[]): number[] {
+		const buffers = outputs.map((vout) => {
+			const value = Buffer.alloc(8);
+			value.writeBigUInt64LE(BigInt(vout.value), 0);
+			const scriptLen = Buffer.from([vout.script.length]);
+			return Buffer.concat([value, scriptLen, vout.script]);
+		});
+		return Array.from(Buffer.concat(buffers));
+	}
```

Comment:
let's move all serialize methods to a separate module, as module functions. They should not be part of this class.